wiki/Ranking_Criteria/en.md

```diff
@@ -30,6 +30,7 @@ Important to understand before using:
 ### Rules
 
 - **No two hit objects can be placed on the same tick.** This includes hit circles and the durations of sliders and spinners. osu!mania difficulties are exempt from this.
+- **No two inputs can be less than 20 milliseconds apart.** osu!mania and osu!catch difficulties are exempt from this.
```

Comment:
Are we sure we want this at 20 ms instead of the 10 ms enforced by AiMod/MV? The thread seems to suggest the change *needs* to happen despite community backlash, but after re-reading both the thread and internal discussions, I do not see the reason why.

As far as I understood this is a diff-calc related issue, which I have since confirmed is tracked and worked on by the performance team. Apart from that, we have found no issues with keeping the current sanity limit of 10 ms.

So I feel like there may be a misunderstanding here.

Reply:
Definitely should be understood before making a call.

Can you cc in the performance team that you have discussed this with, and link an issue/tracking thread?

Reply:
I asked @apollo-dw on 2022-04-22 about getting an issue opened, who replied simply saying that it is already tracked and known in the PP dev server, and that he'd try getting a fix out in the next wave of changes. That is all I know.

Reply:
Yup, this is tracked in the PP server (there are no relevant github issues / other threads to my knowledge, if wanted). In my eyes, the SR inflation issue is going to be addressed as soon as the diff-calc refactor stuff happens that makes it much more feasible. I'd be hesitant to say that the diff-calc issue should be taken into account with deciding the amount of milliseconds here

Reply:
i was under the impression that 10ms would be too little and 20ms was a solid minimum based on discussions that happened the day this issue came up. i'm not an authority on which should be chosen since this is a technical concern, but from a mapping side, 10ms wouldn't be a problem imo

Reply:
As long as the framerate limited input is per-key it should be fine at 10 ms I think (ignoring diffcalc), since the idea is to press both buttons at the same time. If it queues them up so the next input comes at the next frame, it could be a problem. 

But yeah we need more details about how the frame limited input works to make an informed decision here I think.

Reply:
Which existing beatmaps have objects closer than 20ms? Would be good to test them out on both stable and lazer to see how things behave.

Reply:
none ranked, but the reason this rule is being considered is https://osu.ppy.sh/beatmapsets/1691083#osu/3455732 (13ms)

Reply:
After discussion with everyone that has any knowledge surrounding these issue, 10ms sounds fine, except for one special case: After a slider end, the next object needs to be >20ms.

How that is phrased is up for discussion.

Reply:
split into two rules, how does this sound?
`- **There must be at least 10 milliseconds between a hit circle and the next object.** osu!mania and osu!catch difficulties are exempt from this.`
`- **There must be at least 20 milliseconds between the end of a slider and the next object.** osu!mania and osu!catch difficulties are exempt from this.`